src/System.Private.Reflection.Core/src/System/Reflection/Runtime/TypeInfos/RuntimeTypeInfo.cs

```diff
@@ -192,9 +193,69 @@ public sealed override MemberInfo[] GetDefaultMembers()
             return defaultMemberName != null ? GetMember(defaultMemberName) : Array.Empty<MemberInfo>();
         }
 
-        public sealed override InterfaceMapping GetInterfaceMap(Type interfaceType)
+        public sealed override InterfaceMapping GetInterfaceMap(Type ifaceType)
+        {
+            if (IsGenericParameter)
+                throw new InvalidOperationException(SR.Arg_GenericParameter);
+
+            if (ifaceType is null)
+                throw new ArgumentNullException(nameof(ifaceType));
+
+            RuntimeTypeInfo ifaceRtType = ifaceType as RuntimeTypeInfo;
+
+            if (ifaceRtType == null)
+                throw new ArgumentException(SR.Argument_MustBeRuntimeType, nameof(ifaceType));
+
+            RuntimeTypeHandle ifaceRtTypeHandle = ifaceRtType.TypeHandle;
+
+            Debug.Assert(ifaceType.IsInterface);
+            Debug.Assert(!IsInterface);
+            VerifyInterfaceIsImplemented(ifaceRtTypeHandle);
+
+            // SZArrays implement the methods on IList`1, IEnumerable`1, and ICollection`1 with
+            // SZArrayHelper and some runtime magic. We don't have accurate interface maps for them.
+            if (IsSZArray && ifaceType.IsGenericType)
+                throw new ArgumentException(SR.Argument_ArrayGetInterfaceMap);
+
+            int ifaceInstanceMethodCount = RuntimeAugments.GetNumVirtuals(ifaceRtTypeHandle);
+
+            InterfaceMapping im;
+            im.InterfaceType = ifaceType;
+            im.TargetType = this;
+            im.InterfaceMethods = new MethodInfo[ifaceInstanceMethodCount];
+            im.TargetMethods = new MethodInfo[ifaceInstanceMethodCount];
+
+            var ifaceMethods = ifaceRtType.GetMethods();
+            Debug.Assert(ifaceMethods.Length == ifaceInstanceMethodCount);
+
+            for (int i = 0; i < ifaceInstanceMethodCount; i++)
+            {
+                im.InterfaceMethods[i] = ifaceMethods[i];
```

Comment:
Looking at this right now, i could also just set `im.InterfaceMethods = ifaceRtType.GetMethods();`